src/charts/people-affected-by-crisis-map.js

```diff
@@ -0,0 +1,165 @@
+    // eslint-disable-next-line array-callback-return
+    processedCountryData.map((stream) => {
+      if (stream.Country_name === country) {
+        countryData[stream.variable] = stream.value;
+      }
+    });
+    data.push(countryData);
+  });
+
+  return data;
+};
+
+const dataInjectedGeoJson = (jsonData, groupedData) =>
+  jsonData.map((feature) => {
+    const matchingCountryData = groupedData.find((countryData) => countryData.name === feature.properties.name);
+    if (matchingCountryData) {
+      // eslint-disable-next-line no-param-reassign
+      feature.properties = {
+        ...feature.properties,
+        ...matchingCountryData,
+      };
+    }
+
+    return feature;
+  });
+
+const getColor = (score) => {
+  switch (score) {
+    case '5':
+      return '#7F1850';
+    case '4':
+      return '#AD1156';
+    case '3':
+      return '#D64279';
+    case '2':
+      return '#E4819B';
+    case '1':
+      return '#F6B9C2';
+    default:
+      return '#E6E1E5';
+  }
+};
+
+const highlightFeature = (e) => {
+  const layer = e.target;
+
+  layer.setStyle({
+    fillColor: 'yellow',
+  });
+
+  if (!window.L.Browser.ie && !window.L.Browser.opera && !window.L.Browser.edge) {
+    layer.bringToFront();
+  }
+  // Bind popup to layer
+  layer.bindPopup(layer.feature.properties.name).openPopup();
+};
+
+const onLegendAdd = () => {
+  const div = window.L.DomUtil.create('div', 'info legend');
+  const legendData = [
+    { score: '5', label: '5 - Very High' },
+    { score: '4', label: '4 - High' },
+    { score: '3', label: '3 - Medium' },
+    { score: '2', label: '2 - Low' },
+    { score: '1', label: '1 - Very Low' },
+  ];
+
+  const legendContent = legendData
+    .map((data) => `<i style="background:${getColor(data.score)}"></i><label>${data.label}</label>`)
+    .join('');
+  div.innerHTML = legendContent;
+
+  return div;
+};
+
+function renderPeopleAffectedByCrisisLeaflet() {
+  window.DICharts.handler.addChart({
+    className: 'dicharts--gha-people-affected-by-crisis-leaflet',
+    echarts: {
+      onAdd: (chartNodes) => {
+        Array.prototype.forEach.call(chartNodes, (chartNode) => {
+          const dichart = new window.DICharts.Chart(chartNode.parentElement);
+          const map = window.L.map(chartNode).setView([20, -0.09], 2);
+          const variable = 'Severity_score';
+          let geojsonLayer;
+
+          // Legend
+          const legend = window.L.control({ position: 'topright' });
+          legend.onAdd = onLegendAdd;
+          legend.addTo(map);
+          dichart.showLoading();
+          fetch(MAP_FILE_PATH)
+            .then((response) => response.json())
+            .then((jsonData) => {
+              const geojsonData = jsonData.features;
+              fetchCSVData(CSV_PATH).then((data) => {
+                const processedCountryNameData = matchCountryNames(data, geojsonData);
+                const countries = [...new Set(processedCountryNameData.map((stream) => stream.Country_name))];
```

Comment:
Don't use the spread syntax here ... for some reason, the result isn't what you expect on production ... even using polyfils (which we already are) failed to fix this ... use `Array.from(...)`